### PR TITLE
Let's build a weather bot! Part one

### PR DESCRIPTION
**Pull request by tigercosmos**
r? @aszx87410  @kdchang

### CODE REVIEW

File: source/_posts/lets-build-weather-bot-1.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+理論上做法應該有好多種，這邊分享我的作法，不過可能不是最聰明的，我在 Heroku 建立兩個 APP，一個是給 Line 另個給 Messenger，然後兩個 Heroku 都綁定 Github repo 的 master：
+
+```js
+const {
+    LineBot,
+    MessengerBot
+} = require('Bottender');
+
+const bot = (process.env.chatbotPlatform == 'messenger') ?
```
Comment (ArvinH) [inline source/_posts/lets-build-weather-bot-1.md:98]: LGTM 👍 
Just curious about where does `process.env.chatbotPlatform` come from? and why not use `===` here?

**Review by ArvinH — APPROVED** [review]

File: source/_posts/lets-build-weather-bot-1.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+const server = createServer(bot);
+
+server.listen(5000, () => {
+  console.log('server is running on 5000 port...');
+});
+```
+
+這樣寫完就是最簡單的 Hello World Bot 了，相較於上面幾篇我說只需要看概念就好的教學，是不是簡約多了？
+
+參數設定的地方，可以看到 `process.env.channelSecret` 這樣的寫法，這是代表我們不會把真的 Secret 放在程式碼中，所以我們讓程式從環境變數中取得，環竟變數通常可以在你架設伺服器的服務商網站設定，以 Heroku 為例的話，建立一個 App 之後，可以在 `Setting > Config Vars` 中設定：
```
Comment (aszx87410) [inline source/_posts/lets-build-weather-bot-1.md]: 環竟變數 => 環境變數

**Review by aszx87410 — APPROVED** [review]
LGTM

BTW, I want to share my experience on building a chatbot. I have implemented a simple chatbot before, and I found that the most difficult thing is debugging. It's hard to debug because in general you can't run your chatbot service on your local(webhook url requires a https domain). 

Fortunately I found an app called `ngrok`, it creates a tunnel from your localhost to an https subdomain, so that you can test your chatbot on your local. You can give it a try.

File: source/_posts/lets-build-weather-bot-1.md (hunk 1 above)
Comment (tigercosmos) [reply source/_posts/lets-build-weather-bot-1.md:98]: 是環境變數 我補充在文章了
此外 == 或 === 在這邊沒太大差別 因為讀環境變數都是字串 不會有型別上的意外

**Comment by tigercosmos** [comment]
@aszx87410  thanks. All fix.
Seems that bottender supports local test. I would dig into this. I will also try `ngrok`.

File: source/_posts/lets-build-weather-bot-1.md (hunk 1 above)
Comment (ArvinH) [reply source/_posts/lets-build-weather-bot-1.md:98]: Got it, you set up two Heroku Apps and give them the different environment setting.

**Comment by tigercosmos** [comment]
This could be merged I thought?

**Comment by tigercosmos** [comment]
thanks

**Pull request merged**